azurerm/resource_arm_application_gateway.go

```diff
@@ -0,0 +1,1691 @@
+package azurerm
+
+import (
+	"bytes"
+	"encoding/base64"
+	"fmt"
+	"log"
+	"net/http"
+	"path"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/arm/network"
+	"github.com/hashicorp/errwrap"
+	"github.com/hashicorp/terraform/helper/hashcode"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+	"github.com/jen20/riviera/azure"
+)
+
+func resourceArmApplicationGateway() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmApplicationGatewayCreate,
+		Read:   resourceArmApplicationGatewayRead,
+		Update: resourceArmApplicationGatewayCreate,
+		Delete: resourceArmApplicationGatewayDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"location": {
+				Type:      schema.TypeString,
+				Required:  true,
+				ForceNew:  true,
+				StateFunc: azureRMNormalizeLocation,
+			},
+
+			"resource_group_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"sku": {
+				Type:     schema.TypeSet,
+				Required: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:             schema.TypeString,
+							Required:         true,
+							DiffSuppressFunc: ignoreCaseDiffSuppressFunc,
+							ValidateFunc: validation.StringInSlice([]string{
+								string(network.StandardSmall),
+								string(network.StandardMedium),
+								string(network.StandardLarge),
+								string(network.WAFLarge),
+								string(network.WAFMedium),
+							}, true),
+						},
+
+						"tier": {
+							Type:             schema.TypeString,
+							Required:         true,
+							DiffSuppressFunc: ignoreCaseDiffSuppressFunc,
+							ValidateFunc: validation.StringInSlice([]string{
+								string(network.Standard),
+								string(network.WAF),
+							}, true),
+						},
+
+						"capacity": {
+							Type:         schema.TypeInt,
+							Required:     true,
+							ValidateFunc: validation.IntBetween(1, 10),
+						},
+					},
+				},
+				Set: hashApplicationGatewaySku,
```

Comment:
Hey @isaacsgi I have - however I feel this is better answered by of the Terraform Core developers (@apparentlymart / @jbardin) who are in a better position to answer this (I see Martin's already replied)